### ISSUE TITLE
Inline private modifier of delegate should be ok (instead => Style/AccessModifierDeclarations)

### ISSUE THREAD

**Issue by stripedpumpkin** [rubocop/rubocop#13058]
Inline private modifier of `delegate` is necessary for the resulting method to actually be private. But rubocop 
complains with [Correctable] Style/AccessModifierDeclarations: private should not be inlined in method definitions.

Consider the following code:

```ruby
# person.rb

# frozen_string_literal: true

class Person
  def initialize(firstname:)
    @firstname = firstname
  end

  attr_reader :firstname
end

# greeter.rb

# frozen_string_literal: true

class Greeter
  def initialize(person:)
    @person = person
  end

  def call
    "Hello, #{firstname}!"
  end

  private

  attr_reader :person
  private(*delegate(:firstname, to: :person))
end
```
we get the following behavior
```ruby
[1] pry(main)> ann = Person.new(firstname: 'Ann')
=> #<Person:0x00007f5f550e9e60 @firstname="Ann">
[2] pry(main)> greeter = Greeter.new(person: ann)
=> #<Greeter:0x00007f5f564299d0 @person=#<Person:0x00007f5f550e9e60 @firstname="Ann">>
[3] pry(main)> greeter.call
=> "Hello, Ann!"
[4] pry(main)> greeter.firstname
NoMethodError: private method `firstname' called for an instance of Greeter
from (pry):4:in `__pry__'
```
which is the desired behavior.

If we make the suggested change with

```ruby
class Greeter
  def initialize(person:)
    @person = person
  end

  def call
    "Hello, #{firstname}!"
  end

  private

  attr_reader :person
  delegate(:firstname, to: :person)
end
```
the firstname method can be accessed which is not the intention.

## Expected behavior

Rubocop should not raise a Style/AccessModifierDeclarations for inline private modifier of delegation using delegate and autocorrection should not alter the inline privacy modifier.

## Actual behavior

Rubocop raises a Style/AccessModifierDeclarations violation and -a autocorrection will remove the private modifier in a way that does not preserve the privacy of the resulting method.

See below for the full output

## Steps to reproduce the problem

See code above for a functional example, as for my configuration:

```yaml
require:
  - rubocop-capybara
  - rubocop-minitest
  - rubocop-performance
  - rubocop-rails

AllCops:
  TargetRubyVersion: 3.3
  TargetRailsVersion: 7.1
  NewCops: enable
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.65.0 (using Parser 3.3.4.0, rubocop-ast 1.31.3, running on ruby 3.3.4) [x86_64-linux]
  - rubocop-capybara 2.21.0
  - rubocop-minitest 0.35.1
  - rubocop-performance 1.21.1
  - rubocop-rails 2.25.1
```

## Full output

```
root@ab222e695eb2:/some_app# be rubocop app/poros/greeter.rb --debug
For /some_app: configuration from /some_app/.rubocop.yml
configuration from /usr/local/bundle/gems/rubocop-capybara-2.21.0/config/default.yml
Default configuration from /usr/local/bundle/gems/rubocop-1.65.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-capybara-2.21.0/lib/../config/default.yml
configuration from /usr/local/bundle/gems/rubocop-minitest-0.35.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-minitest-0.35.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-performance-1.21.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-performance-1.21.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.25.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.25.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /some_app/app/poros/greeter.rb
Loading cache from /root/.cache/rubocop_cache/bf1b41c785ed7c21e82cef7ae46ec2c3de7e00d7/9741b73e32a55fa25bfd0632cfcd5496c34a0c79/9ac823e6c338faf8d6d0a914f7282a9aed8e102b
C

Offenses:

app/poros/greeter.rb:15:3: C: [Correctable] Style/AccessModifierDeclarations: private should not be inlined in method definitions.
  private(*delegate(:firstname, to: :person))
  ^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.455986370001483 seconds

```

**Comment by dvandersluis** [comment]
This is kind of an extension of #13213, I think splats should be accepted on methods as well.

FWIW, if you're using Rails 6+ (see https://www.bigbinary.com/blog/rails-6-adds-private-option-to-delegate-method) you could change your delegate to

```ruby
delegate(:firstname, to: :person, private: true)
```
